<launch>
	<!-- Data parameters -->
	<arg name="data_file_base_path" default="$(find smarc_algal_bloom_tracking)/data"/>
	<arg name="output_data_path" default="$(find smarc_algal_bloom_tracking)/output"/>
	<arg name="data_downs_scale_factor" value="100.0" />
	<arg name="offset_gps" value="True"/>
	<arg name="data_rotate_angle" value="0.0" />
	<arg name="sampling_time" value="1.0" />
	<arg name="delta_ref" value="7.45" />
	<arg name="robot_name" default="sam"/>
	<arg name="gps_topic" default="/$(arg robot_name)/core/gps" />


    <node name="simulated_chlorophyll_sampler" pkg="smarc_algal_bloom_tracking" type="simulated_chlorophyll_sampler.py" output="screen">
		<param name="data_file_base_path" value="$(arg data_file_base_path)"/>
		<param name="output_data_path" value="$(arg output_data_path)"/>
		<param name="data_downs_scale_factor" value="$(arg data_downs_scale_factor)"/>
		<param name="offset_gps" value="$(arg offset_gps)"/>
		<param name="data_rotate_angle" value="$(arg data_rotate_angle)"  />
		<param name="sampling_time" value="$(arg sampling_time)"  />
		<param name="delta_ref" value="$(arg delta_ref)"  />

		<!-- SIMULATIONS -->
		<!-- ============================================================================= -->

		<!-- Downscale 100 : Full path -->
		<!-- <param name="starting_lat" value="61.4193"/>
		<param name="starting_lon" value="20.7909"/> -->

		<!-- Downscale-100:middle down ish-->
		<param name="starting_lat" value="61.4211"/>
		<param name="starting_lon" value="20.7919"/>

		<!-- ============================================================================= -->

		<!-- OLD -->
		<!-- <param name="starting_lat" value="61.42112"/>
		<param name="starting_lon" value="20.79051"/> -->

		<!-- Downscale-100:middle ish-->
		<!-- <param name="starting_lat" value="61.4218"/>
		<param name="starting_lon" value="20.7923"/> -->

		<!-- Downscale-100:middle down ish-->
		<!-- <param name="starting_lat" value="61.4211"/>
		<param name="starting_lon" value="20.7919"/> -->

		<!-- Longer line -->
		<!-- <param name="starting_lat" value="61.4215"/>
		<param name="starting_lon" value="20.7926"/> -->

		<!-- Downscale-100:bottom left -->
		<!-- <param name="starting_lat" value="61.4193"/>
		<param name="starting_lon" value="20.7909"/> -->

		<!-- Downscale-1:middle -->
		<!-- <param name="starting_lat" value="61.6568"/>
		<param name="starting_lon" value="20.988"/> -->

		<!-- Downscale-1:bottom -->
		<!-- <param name="starting_lat" value="61.4916"/>
		<param name="starting_lon" value="20.9245"/> -->

		<!-- Positional information -->
		<param name="gps_topic" value="$(arg gps_topic)" />

    </node>

	<node name="plot_live_grid" pkg="smarc_algal_bloom_tracking" type="plot_live_grid.py" output="screen">
		<param name="data_file_base_path" value="$(arg data_file_base_path)"/>
		<param name="output_data_path" value="$(arg output_data_path)"/>
		<param name="data_downs_scale_factor" value="$(arg data_downs_scale_factor)"/>
		<param name="offset_gps" value="$(arg offset_gps)"/>
		<param name="data_rotate_angle" value="$(arg data_rotate_angle)"  />
		<param name="sampling_time" value="$(arg sampling_time)"  />
		<param name="delta_ref" value="$(arg delta_ref)"  />

		<!-- Positional information -->
		<param name="gps_topic" value="$(arg gps_topic)" />
    </node>

</launch>